src/layouts/components/CardContainer.js

```diff
@@ -2,31 +2,82 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 
+import { media } from '../../theme/globalStyle'
+
 const StyledCardContainer = styled.div`
   display: grid;
   grid-template-columns: repeat(12, 1fr);
   grid-template-rows: auto;
   margin: 1.5rem 0 3rem 0;
+  ${media.desktop`
+    margin: 0;
+  `} ${media.phone`
+    margin: 0.5rem 0 1rem 0;
+  `};
 `
 
 const StyledStories = styled.div`
-  grid-column: 3 / span 8;
+  grid-column: 2 / span 10;
   display: grid;
   grid-template-columns: ${props => `repeat(${props.cols}, 1fr)`};
   grid-template-rows: auto;
   grid-gap: 0.5rem;
   border: 2px solid ${props => props.theme.primary.dark};
-  padding: 1rem;
+  border-radius: 2px;
+  padding: 0.6rem;
+  & > div:last-child {
```

Comment:
This looks pretty neat, I'm guessing it's taking the `props` if it's on the last column, then using display `flex` instead of `grid` correct?

Reply:
I'm afraid I made it quite complicated :blush: 
`CardContainer` needs 3 props. 1 of them is required (`cols`) and 2 are optional (`cards`, `story`).

I added story because for the `StoryCard` I combined **grid** and **flex** layouts. (I found more easy to use `flex` with changing viewports - less CSS to add :smile: ) and `ProjectCard` uses only **flex**. `story` prop is a boolean to tell that I'm going to use the `CardContainer` with `StoryCards`.

The aim of `cols` props is to tell `CardContainer` how many columns of the **Grid** to build (and how many cards to show).
However, if the number of columns is **greater then 3 and odd** we will end up with 1 empty slot on a tablet screen (for cards = 5, for other values of cards the picture will be different):
 
![image](https://user-images.githubusercontent.com/25195616/34604336-0bd42a64-f210-11e7-93fb-5106288ebba9.png)

It forced me to add 1 more card and hide it till the tablet viewport is reached. This is the source of `cards` prop. It is a real number of card which we pass to the `addCards()` helper function to generate cards.
   
I know this solution is not very universal but it works for all the numbers of `Cards` we have in our app. :upside_down_face: